src/tv2-common/videoSwitchers/TriCaster.ts

```diff
@@ -175,17 +191,79 @@ export class TriCaster extends VideoSwitcherImpl {
 		)
 	}
 
-	public getDveTimelineObjects(_properties: DveProps): TSR.TSRTimelineObj[] {
-		throw new Error('Method not implemented.')
-	}
-
 	public updateUnpopulatedDveBoxes(
 		_timelineObject: TSR.TSRTimelineObj,
 		_input: number | SpecialInput
 	): TSR.TSRTimelineObj {
 		throw new Error('Method not implemented.')
 	}
 
+	public getDveTimelineObjects(dveProps: DveProps): TSR.TSRTimelineObj[] {
+		return [
+			literal<TSR.TimelineObjTriCasterME>({
+				id: '',
+				enable: dveProps.enable ?? { start: 0 },
+				layer: TRICASTER_LAYER_PREFIX + SwitcherDveLLayer.DveBoxes,
+				priority: 1,
+				content: {
+					deviceType: TSR.DeviceType.TRICASTER,
+					type: TSR.TimelineContentTypeTriCaster.ME,
+					me: literal<TSR.TriCasterMixEffectInEffectMode>({
+						transitionEffect: 8,
+						layers: this.generateDveBoxLayers(dveProps.content.boxes),
+						keyers: this.generateOverlayKeyer()
+					})
+				}
+			})
+		]
+	}
+
+	private generateDveBoxLayers(boxes: any[]): Partial<Record<TSR.TriCasterLayerName, TSR.TriCasterLayer>> {
+		return {
+			a: boxes[0].enabled ? this.generateDveBoxLayout(boxes[0]) : this.generateInvisibleBoxLayer(),
+			b: boxes[1].enabled ? this.generateDveBoxLayout(boxes[1]) : this.generateInvisibleBoxLayer(),
+			c: boxes[2].enabled ? this.generateDveBoxLayout(boxes[2]) : this.generateInvisibleBoxLayer(),
+			d: boxes[3].enabled ? this.generateDveBoxLayout(boxes[3]) : this.generateInvisibleBoxLayer()
+		}
+	}
+
+	private generateDveBoxLayout(box: any): TSR.TriCasterLayer {
+		return {
+			input: this.getInputName(box.source),
+			positioningAndCropEnabled: true,
```

Comment:
Is this ever false?

Reply:
Not in the current implementation of generating DVEs, but it's required to be true in TriCaster for us to manipulate position and the it's optional in the interface